lib/src/builder_implementation.dart

```diff
@@ -3902,125 +3877,97 @@ ${imports.join('\n')}
+import "package:reflectable/src/reflectable_builder_based.dart" as r;
 // ignore:unused_import
-import "package:reflectable/reflectable.dart" show isTransformed;
+import "package:reflectable/reflectable.dart" as r show Reflectable;
 
 $code
 
 final _memberSymbolMap = ${world.generateSymbolMap()};
 
 initializeReflectable() {
-  if (!isTransformed) {
-    throw new UnsupportedError(
-        "The transformed code is running with the untransformed "
-        "reflectable package. Remember to set your package-root to "
-        "'build/.../packages'.");
-  }
   r.data = _data;
   r.memberSymbolMap = _memberSymbolMap;
 }
 """;
     if (_formatted) {
-      DartFormatter formatter = new DartFormatter();
+      DartFormatter formatter = DartFormatter();
       result = formatter.format(result);
     }
     return result;
   }
 
-  /// Performs the transformation which eliminates all imports of
-  /// `package:reflectable/reflectable.dart` and instead provides a set of
-  /// statically generated mirror classes.
-  Future apply(Transform transform, List<String> entryPoints, bool formatted,
-      List<WarningKind> suppressedWarnings) async {
-    // The type argument in the return type is omitted because the
-    // documentation on barback and on transformers do not specify it.
-
-    _logger = transform.logger;
+  /// Perform the build which produces a set of statically generated
+  /// mirror classes, as requested using reflectable capabilities.
+  String buildMirrorLibrary(
+      Resolver resolver,
+      AssetId inputId,
+      AssetId generatedLibraryId,
+      LibraryElement inputLibrary,
+      List<LibraryElement> visibleLibraries,
+      bool formatted,
+      List<WarningKind> suppressedWarnings) {
     _formatted = formatted;
     _suppressedWarnings = suppressedWarnings;
 
-    Asset asset = await transform.primaryInput;
-    assert(asset != null); // Every transform has a primary asset.
-    AssetId currentEntryPoint; // Null means not found.
+    // The [_resolver] provides all the static information.
+    _resolver = resolver;
+    _libraries = visibleLibraries;
 
-    for (String entryPoint in entryPoints) {
-      Glob glob = new Glob(entryPoint);
-      if (glob.matches(asset.id.path)) {
-        currentEntryPoint = asset.id;
-        break;
-      }
+    for (LibraryElement library in _libraries) {
+      if (library.name != null) _librariesByName[library.name] = library;
     }
-    // If the given [asset] is not an entry point: skip.
-    if (currentEntryPoint == null) return;
+    print(_librariesByName); // DEBUG
+    LibraryElement reflectableLibrary =
+        _librariesByName["reflectable.reflectable"];
+
+    if (reflectableLibrary == null) {
+      // Stop and let the original source pass through without changes.
+      log.info("Ignoring entry point $inputId that does not "
+          "include the library 'package:reflectable/reflectable.dart'");
+      if (const bool.fromEnvironment("reflectable.pause.at.exit")) {
```

Comment:
cc @jakemac53 

Note that we don't know what environment variables you read during the build and won't invalidate the output if a new build is run with a different environment. Is it possible to put this argument in the `BuilderOptions` and configure with `build.yaml` instead? If it's changed we'll know to rerun the appropriate build steps.

Reply:
Those things are only used for controlling debug-ish output, so it shouldn't matter for the decision about whether or not to regenerate code.